flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerTest.java

```diff
@@ -385,14 +385,14 @@ public void testStopWorkerAfterRegistration() throws Exception {
 			final CompletableFuture<Void> startContainerAsyncFuture = new CompletableFuture<>();
 			final CompletableFuture<Void> stopContainerAsyncFuture = new CompletableFuture<>();
 
-			testingYarnAMRMClientAsync.setGetMatchingRequestsFunction(ignored ->
+			testingYarnAMRMClientAsyncBuilder.setGetMatchingRequestsFunction(ignored ->
```

Comment:
From my understanding, we are sure about that. The `build` function is triggered when the ResourceManager is started, to be specific, in `Context#runTest`. As we now set the builders and trigger `Context#runTest` sequentially in the testing main thread, we could ensure the builders and `containerResource` will always be seen by `ResourceManager`. Do I understand it correctly?

Reply:
I think you are probably right because the first read of the builder in the `RpcEndpoint's` main thread happens after we set up the builder. Hence, nothing should be cached. Moreover, if this should be a problem, then it should soon surface on AZP.